postgres/datadog_checks/postgres/statement_samples.py

```diff
@@ -295,51 +319,50 @@ def _collect_statement_samples(self):
 
     def _can_explain_statement(self, obfuscated_statement):
         if obfuscated_statement.startswith('SELECT {}'.format(self._explain_function)):
-            return False
+            return False, {'code': FailedExplainReason.no_plans_possible.value}
         if obfuscated_statement.startswith('autovacuum:'):
-            return False
+            return False, {'code': FailedExplainReason.no_plans_possible.value}
         if obfuscated_statement.split(' ', 1)[0].lower() not in VALID_EXPLAIN_STATEMENTS:
-            return False
-        return True
+            return False, {'code': FailedExplainReason.invalid_statement.value}
+        return True, None
 
     def _get_db_explain_setup_state(self, dbname):
-        # type: (str) -> DBExplainSetupState
         try:
             self._get_db(dbname)
         except (psycopg2.DatabaseError, psycopg2.OperationalError) as e:
             self._log.warning(
                 "cannot collect execution plans due to failed DB connection to dbname=%s: %s", dbname, repr(e)
             )
-            return DBExplainSetupState.failed_connect
+            return DBExplainSetupState.failed_connect, e
 
         try:
             result = self._run_explain(dbname, EXPLAIN_VALIDATION_QUERY, EXPLAIN_VALIDATION_QUERY)
         except psycopg2.errors.InvalidSchemaName as e:
             self._log.warning("cannot collect execution plans due to invalid schema in dbname=%s: %s", dbname, repr(e))
-            return DBExplainSetupState.invalid_schema
+            return DBExplainSetupState.invalid_schema, e
         except psycopg2.DatabaseError as e:
             # if the schema is valid then it's some problem with the function (missing, or invalid permissions,
             # incorrect definition)
             self._log.warning("cannot collect execution plans in dbname=%s: %s", dbname, repr(e))
-            return DBExplainSetupState.failed_function
+            return DBExplainSetupState.failed_function, e
 
         if not result:
-            return DBExplainSetupState.invalid_result
+            return DBExplainSetupState.invalid_result, None
 
-        return DBExplainSetupState.ok
+        return DBExplainSetupState.ok, None
 
     def _get_db_explain_setup_state_cached(self, dbname):
-        # type: (str) -> DBExplainSetupState
```

Comment:
This is a type hint comment: https://realpython.com/lessons/type-comments/

Reply:
Yeah, I initially removed this because when I changed the return signature, the format linter would fail with the more custom return type i.e `Tuple(DBExplainSetupState, Dict)` and if I remember correctly, Python versions < 2.7 does not support the imported type `Dict`. I'll revisit this again to make sure.

Reply:
Well it's a different type now so it should fail. It's now a tuple of two elements: `Tuple[str, exception]`